java/algorithms/BellmanFord-FunSat01/Main.java

```diff
@@ -0,0 +1,109 @@
+ * <p/>
+ * 2. Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution.
+ * <p/>
+ * 3. Neither the name of the University of California, Berkeley nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ * <p/>
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @author Sudeep Juvekar <sjuvekar@cs.berkeley.edu>
+ * @author Jacob Burnim <jburnim@cs.berkeley.edu>
+ */
+public class Main {
+
+  static final int INFINITY = 1000000;
+
+  static int[] runBellmanFord(int N, int D[], int src) {
+    // Initialize distances.
+    int dist[] = new int[N];
+    boolean infinite[] = new boolean[N];
+    for (int i = 0; i < N; i++) {  // V+1 branches
+      dist[i] = INFINITY;
+      infinite[i] = true;
+    }
+    dist[src] = 0;
+    infinite[src] = false;
+
+    // Keep relaxing edges until either:
+    //  (1) No more edges need to be updated.
+    //  (2) We have passed through the edges N times.
+    int k;
+    for (k = 0; k < N; k++) {  // V+1 branches
+      boolean relaxed = false;
+      for (int i = 0; i < N; i++) {  // V(V+1) branches
+        for (int j = 0; j < N; j++) {  // V^2(V+1) branches
+          if (i == j) continue;  // V^3 branches
+          if (!infinite[i]) {  // V^2(V-1) branches
+            if (dist[j] > dist[i] + D[i*N+j]) {  // V^2(V-1) branches
+              dist[j] = dist[i] + D[i*N+j];
+              infinite[j] = false;
+              relaxed = true;
+            }
+          }
+        }
+      }
+      if (!relaxed)  // V branches
+        break;
+    }
+
+    // Check for negative-weight egdes.
+    if (k == N) {  // 1 branch
+      // We relaxed during the N-th iteration, so there must be
+      // a negative-weight cycle.
+    }
+
+    // Return the computed distances.
+    return dist;
+  }
+
+  public static void main(String[] args) {
+    final int V = Verifier.nondetInt();
+    if (V <= 0) return; // change with Verifier.assume?
+
+    final int D[] = new int[V*V];
```

Comment:
I guess the change `INFINITY = Integer.MAX_VALUE` should resolve this:
> nit-pick:
`INFINITY < V` might occur using `Verifier.nondetInt()`

Adding the assumptions I've mentioned in another comment should resolve the overflow issue I think: 
- assume that V is at most 1e6, which should still be almost impossible to verify using bounded methods and cause no overflows in the calculations
- assume the same with the initialization of D at line 98

> Specifying the reachability of certain Standard Java-Exceptions might be one option report this overflow or we might add an assert statement that ensures no overflow before the int.

We have assumed that the verifier should check the reachability of these standard exceptions. If this is not suitable, maybe we can wrap the potentially unsafe code with try-catch blocks, with assert false; in the catch statement?

Reply:
It is my understanding that currently only explicit assert violations are checked in this category according to the property file. Using a try-catch blog sounds legit to me. But I might be wrong. 
Otherwise it might be possible to define a new property for checking the standard errors on a program and use such a property along with this file. I am not sure, what would be the better approach.

Eventually @peterschrammel or @dbeyer might give some guidance for the better solution here.